src/main/java/com/kk/strong/security/SecurityConfig.java

```diff
@@ -0,0 +1,59 @@
+package com.kk.strong.security;
+
+import com.kk.strong.model.UserRole;
+import lombok.AllArgsConstructor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.config.annotation.authentication.configuration.AuthenticationConfiguration;
+import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+@Configuration
+@EnableWebSecurity
+@EnableGlobalMethodSecurity(securedEnabled = true)
+@AllArgsConstructor
+public class SecurityConfig {
+
+    protected static final int ACCESS_TOKEN_EXPIRY_SECONDS = 60 * 60;
+    protected static final String TOKEN_SECRET = "token_secret";
+
+    private final AuthenticationConfiguration authenticationConfiguration;
+
+    @Bean
+    public PasswordEncoder passwordEncoder() {
+        return new BCryptPasswordEncoder();
+    }
+
+    @Bean
+    public AuthenticationManager authenticationManager(AuthenticationConfiguration authenticationConfiguration) throws Exception {
+        return authenticationConfiguration.getAuthenticationManager();
+    }
+
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity httpSecurity) throws Exception {
+
+        CustomAuthenticationFilter customAuthenticationFilter = new CustomAuthenticationFilter(authenticationManager(authenticationConfiguration));
+        customAuthenticationFilter.setFilterProcessesUrl("/login");
+
+        httpSecurity.csrf().disable();
+        httpSecurity.sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS);
+
+        httpSecurity.authorizeRequests().antMatchers("/login").permitAll();
+        httpSecurity.authorizeRequests().antMatchers("/**").hasAuthority(UserRole.REGULAR_USER.name());
```

Comment:
in this line you allow `REGULAR_USER` to create and view users, you should probably block this activity and allow only `PREMIUM_USER` or event new role `ADMIN` to create and retrieve users